internal/brokers/broker_test.go

```diff
@@ -277,10 +272,10 @@ func TestCancelIsAuthorized(t *testing.T) {
 	}
 }
 
-func newBrokerForTests(t *testing.T) (b brokers.Broker, cfgPath string) {
+func newBrokerForTests(t *testing.T, cfgDir, wantResponse string) (b brokers.Broker) {
```

Comment:
It makes sense as well. I switched it in order to make it more aligned with the one I created for the PAM tests. I thought it would be more straightforward to understand.

Reply:
Honestly, I think this way is still better (even though is a bit more verbose for the caller), to avoid having to call `filepath.Dir` on some tests in the `manager_test.go`, like:
```Go
cfgDir := t.TempDir()
b := newBrokerForTests(t, cfgDir, "does not matter")
m, err := brokers.NewManager(context.Background(), nil, brokers.WithBrokerCfgDir(cfgDir))
```
It does look easier to understand, no?

Reply:
Yeah, but then the function flow gets weird, no? IMHO, it should either return the path to the broker-specific config file or not return anything else besides the broker object. It feels weird to have the function returning the broker and the directory of the config file (rather than the config path itself). I know this is just a private helper for the tests, but I don't think it hurts having to call `t.TempDir()` before going inside the function.